### ISSUE TITLE
ICE TCP: panic: send on closed channel

### ISSUE THREAD

**Issue by jech** [pion/webrtc#1355]
While experimenting with the ICE TCP support, I got me a panic:
```
panic: send on closed channel

goroutine 239 [running]:
github.com/pion/ice/v2.(*tcpPacketConn).handleRecv(0x40006b7450, 0x4000401b90, 0x70, 0x70, 0x65d500, 0x40005241e0, 0x0, 0x0)
        /home/jch/go/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.9/tcp_packet_conn.go:114 +0x124
github.com/pion/ice/v2.(*tcpPacketConn).startReading(0x40006b7450, 0x667780, 0x40001040c0)
        /home/jch/go/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.9/tcp_packet_conn.go:100 +0xc0
github.com/pion/ice/v2.(*tcpPacketConn).AddConn.func1(0x40005c8000, 0x70, 0x2000, 0x40006b7450, 0x667780, 0x40001040c0)
        /home/jch/go/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.9/tcp_packet_conn.go:76 +0x6c
created by github.com/pion/ice/v2.(*tcpPacketConn).AddConn
        /home/jch/go/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.9/tcp_packet_conn.go:71 +0x2e0
```

**Comment by jeremija** [comment]
This is actually an issue in `pion/ice`. It looks like the logic in `tcpPacketConn::handleRecv` is faulty. We check if the channel is closed, but since the mutex is unlocked before, by the time the `select` statement is reached, both channels `closedChan` and `recvChan` might be closed, and the `select` will randomly select a statement to execute:

```go
func (t *tcpPacketConn) handleRecv(pkt streamingPacket) {
	t.mu.Lock()

	recvChan := t.recvChan
	if t.isClosed() {
		recvChan = nil
	}

	t.mu.Unlock()

	select {
	case recvChan <- pkt:
	case <-t.closedChan:
	}
}
```

**Comment by jeremija** [comment]
@jech  Could you try cloning the github.com/pion/ice repo locally, checking out the `issue-260` branch and modifying your go.mod to include:

```
replace github.com/pion/ice/v2 => /path/to/local/pion/ice
```

and see if you can reproduce the panic?

**Comment by jech** [comment]
I'm unable to reproduce the crash with the issue-260 branch.  However, the crash is difficult to trigger in the first place, so perhaps I've just failed to reproduce it.

**Comment by jeremija** [comment]
This should be resolved by #1357. Please reopen if you run into this issue again!
